src/coreclr/jit/llvm.cpp

```diff
@@ -2387,6 +2409,17 @@ void Llvm::lowerCallToShadowStack(GenTreeCall* callNode)
     }
 }
 
+void Llvm::convertLclStructToLoad(GenTreeLclVarCommon* lclNode, ClassLayout* clsLayout)
+{
+    GenTree* lclAddr = _compiler->gtNewLclVarAddrNode(lclNode->GetLclNum(), TYP_STRUCT);
```

Comment:
Seems doubtful this is the corruption cause, but this should be `TYP_I_IMPL`, not `TYP_STRUCT`.

(Or actually omitted -- I believe `gtNewLclVarAddrNode` has `TYP_I_IMPL` as the default parameter already).
```suggestion
    GenTree* lclAddr = _compiler->gtNewLclVarAddrNode(lclNode->GetLclNum());
```

Reply:
Yes, you are right `GenTreeLclVar* gtNewLclVarAddrNode(unsigned lclNum, var_types type = TYP_I_IMPL);`  Thanks